SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/DefaultAnimationLoop.cpp

```diff
@@ -140,11 +140,12 @@ void DefaultAnimationLoop::step(const core::ExecParams* params, SReal dt)
     }
     sofa::helper::AdvancedTimer::stepEnd("UpdateMapping");
 
-    if (!SOFA_NO_UPDATE_BBOX)
+#if SOFA_NO_UPDATE_BBOX == 0
```

Comment:
Hi alex,

If possible, the rules is to do as much as possible
```cpp
if(SOFA_NO_UPDATE_BBOX)
{
    ....
}
```
The rational behind is that the conditional content is still compiled (eg: in the CI) so this ease maintenance and code refactoring while the compiler's optimizer can strip-out the if() when it is false.

Reply:
Since SOFA_NO_UPDATE_BBOX is a preprocessor variable, it makes more sense to me to use a preprocessor condition on it. IDEs would color the code differently on a deactivated code.
If you put the condition in a traditional condition, I am not sure the compiler compiles the code if the condition is false. IMO it sees it is false and does not bother compiling it. But I may be wrong, I don't know it for a fact.
The preprocessor condition would also remove the warning "the condition is always true".
A third alternative is 
`if constexpr (SOFA_NO_UPDATE_BBOX)
{
}`
What do you think about this one?

Honestly, I don't have a strong preference. I don't mind reverting this change